wagtail_localize/views/submit_translations.py

```diff
@@ -0,0 +1,212 @@
+
+        if not isinstance(instance, Page) or instance.get_descendants().count() == 0:
+            self.fields["include_subtree"].widget = forms.HiddenInput()
+
+        self.fields["locales"].queryset = Locale.objects.exclude(
+            id__in=instance.get_translations(inclusive=True).values_list('locale_id', flat=True)
+        )
+
+        # Using len() instead of count() here as we're going to evaluate this queryset
+        # anyway and it gets cached so it'll only have one query in the end.
+        if len(self.fields["locales"].queryset) < 2:
+            self.fields["select_all"].widget = forms.HiddenInput()
+
+
+class TranslationCreator:
+    """
+    A class that provides a create_translations method.
+
+    Call create_translations for each object you want to translate and this will submit
+    that object and any dependencies as well.
+
+    This class will track the objects that have already submitted so an object doesn't
+    get submitted twice.
+    """
+    def __init__(self, user, target_locales):
+        self.user = user
+        self.target_locales = target_locales
+        self.seen_objects = set()
+
+    def create_translations(self, instance, include_related_objects=True):
+        if isinstance(instance, Page):
+            instance = instance.specific
+
+        if instance.translation_key in self.seen_objects:
+            return
+        self.seen_objects.add(instance.translation_key)
+
+        source, created = TranslationSource.get_or_create_from_instance(instance)
+
+        # Add related objects
+        # Must be before translation records or those translation records won't be able to create
+        # the objects because the dependencies haven't been created
+        if include_related_objects:
+            for related_object_segment in source.relatedobjectsegment_set.all():
+                related_instance = related_object_segment.object.get_instance(instance.locale)
+
+                # Limit to one level of related objects, since this could potentially pull in a lot of stuff
+                self.create_translations(related_instance, include_related_objects=False)
+
+        # Set up translation records
+        for target_locale in self.target_locales:
+            # Create translation if it doesn't exist yet
+            translation, created = Translation.objects.get_or_create(
+                source=source,
+                target_locale=target_locale,
+            )
+
+            if created:
+                translation.save_target(user=self.user)
+
+
+class SubmitTranslationView(SingleObjectMixin, TemplateView):
+    template_name = "wagtail_localize/admin/submit_translation.html"
+    title = __("Translate")
+
+    def get_title(self):
+        return self.title
+
+    def get_subtitle(self):
+        return str(self.object)
+
+    def get_form(self):
+        if self.request.method == 'POST':
+            return SubmitTranslationForm(self.object, self.request.POST)
```

Comment:
Could you use `return SubmitTranslationForm(self.object, self.request.POST or None)` instead of if/else?

Reply:
Tried changing this, but it broke the tests (validation wasn't running when it submitted the form without any locales selected)